### ISSUE TITLE
not working with sveltekit

### ISSUE THREAD

**Issue by parence** [JonasKruckenberg/imagetools#318]
Starting from a fresh svelte kit skeleton project (```npm init svelte@next```) and only installing  vite-imagetools I edited:

svelte.cfg.js:
```js
import adapter from '@sveltejs/adapter-auto';
import preprocess from 'svelte-preprocess';              
import { imagetools } from 'vite-imagetools';
 
/** @type {import('@sveltejs/kit').Config} */
const config = {
        // Consult https://github.com/sveltejs/svelte-preprocess
        // for more information about preprocessors
        preprocess: preprocess(),
 
        kit: {
                adapter: adapter()
        },
        vite: {
                plugins: [imagetools()]
        }
};
 
export default config;
```
and src/routes/index.svelte:
```svelte
<script>
        import Image from '../../static/test.jpg?w=400';
        console.log(Image);
</script>
 
<h1>Welcome to SvelteKit</h1>
<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
 
<img src="{Image.src}">
```

This won't show any image. In fact Image.src is undefined since Image is strangely just a string (/static/test.jpg?w=400). If I set just Image to the src attribute the original image is shown without any transformations (i.e. the width is not 400). 

What am I doing wrong here?

**Comment by KocsisGabor** [comment]
I think Image is the src in your case so you should use 

```html
<img src="{Image}">
```


If you need metadata which can return the image as an object then you should import this way:  


```js
import Image from '../../static/test.jpg?w=400&metadata';
```

In the above case Image will be an [object](https://github.com/JonasKruckenberg/imagetools/blob/main/docs/guide/getting-started.md#metadata) with an src attribute so you can use {Image.src} with metadata.

**Comment by parence** [comment]
Thank you for your explanations and suggestions!

I tried
```html
<img src="{Image}">
```
as you suggested and I can confirm that while using this approach the image is displayed. However, the transformations don't seem to be applied (i.e. in the above example image's width is **not** 400, it's still in its original resolution).

I also tried adding `&metadata` and `Image` is still a string (not an object as expected).

Something seems broken in the set up I posted above, which is just bare bones sveltekit + imagetools. But I don't see any errors during build. Is there something else I need to configure?

**Comment by KocsisGabor** [comment]
I have essentially the same setup and I use it the same way (except I use the ?meta shorthand and I use $static/ instead of the relative path).  

Try imagetools({ force: true}) in the vite config, that is a deprecated option which in the past enabled the plugin in dev mode. Or try a different version (I use "vite-imagetools": "^4.0.3").  

Sorry I couldn’t help more, your setup should work.

**Comment by JonasKruckenberg** [comment]
The explanation @KocsisGabor gave it correct, so appending `&metadata` should return you an object with a number of properties (`src` is one of them) while normally you get a string - the url to the transformed image.

If you @parence could share a quick minimal repro of your issue I can have a look.

**Comment by parence** [comment]
Thanks @KocsisGabor, I have tried with `{force: true}` but without any success. I'm also using the same version as you.

@JonasKruckenberg  I have pushed a skeleton project here: https://github.com/parence/imagetools-sveltekit

**Comment by yndotdev** [comment]
I have the same issue as @parence w/ Sveltekit.

**Comment by KocsisGabor** [comment]
I'm sorry @parence I was mistaken when I said I have essentially the same setup, I didn't notice one important difference:  The vite config object should go into the kit object, so it should be:  

```js
const config = {
    preprocess: preprocess(),
    kit: {
        adapter: adapter(),
        vite: {
            plugins: [imagetools()]
        }
    },
};
```

PR would be an overkill for this. Also, I actually only noticed this mistake when I cloned your minimal repo and VSCode underlined that part in the config as a problem, I think it's worth setting up a code editor to do things like that, it can save you a lot of time.  

Maybe the same mistake is present in your config @yndotdev?

**Comment by parence** [comment]
@KocsisGabor you are absolutely right. That solved it for me!
I misunderstood the sveltekit docs concerning the svelte config. Do you mind sharing how you configured your linter? I have eslint configured in my project and it definitely complains when I introduce other errors in my svelte.config.js. However, in this case it remained completely silent.

Thanks!

**Comment by JonasKruckenberg** [comment]
Glad you guys figured it out! 👍🏻

**Comment by KocsisGabor** [comment]
@parence, These kinds of errors are reported by the ts language server on my machine which I believe is turned on by default in VSCode.  

I actually tried to find the relevant config but I just couldn’t, even when I removed everything (tsconfig.json, .eslintrc.cjs, jsconfig) and even turned off everything in vs code settings that seemed relevant it still showed me this error (also the error was present on the skeleton project so I’m sure it’s some setting in VSCode).  
The only other thing I can think of is maybe VSCode needs typescript to be installed globally.

    npm install -g typescript
